### PR TITLE
add custom authorization header support

### PR DESCRIPTION
**Pull request by Nico87LBS**
add an option to specify a custom header to be checked to retrieve the jwt

### CODE REVIEW

File: lib/extract.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ module.exports = function (request, options) {
       auth = request.headers.authorization;
     }
   }
+  else if (typeof options.customHeader === 'string') {
+    if (request.headers[options.customHeader]) {
+      auth = request.headers[options.customHeader];
```
Comment (nelsonic) [inline lib/extract.js]: @Nico87LBS What is the `use-case` for this? please clarify. thanks.

**Comment by Nico87LBS** [comment]
I've just created two tests, which passed, but 3 previous tests are failing (because of the lack of environment variables ?! I don't know)

**Comment by nelsonic** [comment]
Why would the `Cookie` tests be failing...? https://travis-ci.org/dwyl/hapi-auth-jwt2/builds/94433205

**Comment by nelsonic** [comment]
@Nico87LBS this PR has diverged from master as of the `5.3.0` release. 
If you are still keen to get this feature in, please rebase your PR on Master and re-open it. thanks.

**Pull request closed without merging**